llmfoundry/models/layers/attention.py

```diff
@@ -377,17 +401,21 @@ def __init__(
             fc_kwargs['device'] = device
         self.Wqkv = FC_CLASS_REGISTRY[fc_type](
             self.d_model,
-            3 * self.d_model,
+            self.d_model + 2 * self.kv_n_heads * self.head_dim,
             **fc_kwargs,
         )
         # for param init fn; enables shape based init of fused layers
-        fuse_splits = (d_model, 2 * d_model)
+        fuse_splits = [
+            i * self.head_dim
+            for i in range(1, self.n_heads + 2 * self.kv_n_heads)
+        ]
```

Comment:
I've sanity checked this, but confirming this init

Reply:
lgtm